test/mocks/safe_init/mocks.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "envoy/safe_init/manager.h"
+
+#include "common/safe_init/target_impl.h"
+#include "common/safe_init/watcher_impl.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+namespace SafeInit {
+
+/**
+ * MockWatcher is a real WatcherImpl, subclassed to add a mock `ready` method that you can set
```

Comment:
I'm not such a fan of this style TBH. Ideally mocks should be pure, so that when used in some unrelated code, which only needs to use part of the mocked class, e.g. just the `add()`, it only needs the single `EXPECT_CALL` and nothing else. This approach pushes a lot of policy into the mock. Sometimes we do this for convenience, when we don't want a zillion tests reinventing how to mock `ClusterManager`, for example, but in this case I think we could have a purer mock here and override it based on `WatcherImpl` in the specific tests.

Reply:
What I think I should do, actually, is not provide mocks for `Watcher` and `Target` at all, only a `MockManager`. I just need to find a nice way of feeding them mocks as callbacks.

Reply:
Updated: these really aren't mocks, they are real `WatcherImpl` and `TargetImpl` subclassed to make tests less cumbersome to write. Renamed.